orchestra/orchestra_api.py

```diff
@@ -89,24 +89,39 @@ def get_project_information(project_ids):
 
 def create_todos(todos):
     response = _make_api_request('post', 'todo-api',
+                                 headers={'Content-type': 'application/json'},
                                  data=json.dumps(todos))
     return json.loads(response.text)
 
 
-def get_todos(project_id, step_slug=None):
-    if project_id is None:
-        raise OrchestraError('project_id is required')
-    project_param = 'project={}'.format(project_id)
+def build_url_params(project_id, step_slug, **filters):
+    project_param = 'project__id={}'.format(project_id) if project_id else ''
     step_slug_param = '&step__slug={}'.format(
         step_slug) if step_slug is not None else ''
-    query_params = '?{}{}'.format(project_param, step_slug_param)
-
+    additional_filters = '&q={}'.format(json.dumps(filters))
```

Comment:
minor, and feel free to push back: you could do this only if filters is nonempty :)